### PR TITLE
EDU-186 Extract common dev tasks into reusable modules

### PR DESCRIPTION
**Pull request by ahelmberger**
This PR does a little bit more than in the ticket (sorry, again). The idea is to extract the tasks that are the same in elmu, oma and educandu (and other packages), which is probably 95% reusable (and our gulpfiles a little bit lighter). For now I extracted some stuff to a folder called `dev`, we can continue to do this and then create an npm package out of this directory.

Closes https://educandu.atlassian.net/browse/EDU-186

### CODE REVIEW

**Review by laurentiu-ilici — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -109,7 +109,9 @@
   },
   "devDependencies": {
     "@jest/globals": "^27.3.1",
+    "@rauschma/stringio": "^1.4.0",
```
Comment (adrianaluchian) [inline package.json:112]: hehe, Mr 2ality 😎

File: gulpfile.js [hunk 2]
```diff
@@ -523,17 +397,18 @@ export async function release() {
...
+        { responseType: 'json', auth: { username: cliArgs.jiraUser, password: cliArgs.jiraApiKey } }
       );
     } catch (error) {
       console.log(error);
     }
   }
 }
 
-export const up = gulp.series(mongoUp, minioUp, maildevUp);
+export const up = gulp.parallel(mongoUp, minioUp, maildevUp);
```
Comment (adrianaluchian) [inline gulpfile.js:408]: Nice!

File: dev/process.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+  waitForExit() {
+    return new Promise(resolve => {
+      if (this._process) {
+        this._process.once('exit', async () => {
+          await delay(100);
+          resolve();
+        });
+      } else {
+        delay(0).then(resolve);
```
Comment (adrianaluchian) [inline dev/process.js:58]: I'm missing a bit why this needs to be resolved on the next tick 🤔

File: gulpfile.js (hunk 2 above)
Comment (ahelmberger) [reply gulpfile.js:408]: Yeah, no clue why it was `series` before :grin:

File: package.json (hunk 1 above)
Comment (ahelmberger) [reply package.json:112]: Finally he's officially part of the project

**Comment by adrianaluchian** [comment]
Super cool console when you run multiple, with the color coding distinction 👍 
I tested some imports with multiple instances and all worked fine on my end!

File: dev/process.js (hunk 3 above)
Comment (ahelmberger) [reply dev/process.js:58]: Actually no reason except that I don't like resolving `new Promise`s synchronously in the constructor (so it is resolved even before the promise is assigned to a variable). With `delay(0)` we're always waiting now (just shorter in one case). It's a little bit neurotic, I admit.

**Review by adrianaluchian — APPROVED** [review]

**Pull request merged**